### DISCUSSION TITLE
How to use stack property type

### DISCUSSION THREAD

**Discussion by rajevar** [GrapesJS/grapesjs#5435] [Q&A] [answered]
Hi, 

GrapesJS Version:`0.21.3`

we have a custom StyleManager in Vue to render style properties of components. When i use `background` for text-component, it isn't working as expected. Im not using `grapesjs-style-bg` plugin for now and i believe it is only needed to add `color` and `gradient` to the background layer. 

This is how I have defined the `background` property in the SectorManager
![image](https://github.com/GrapesJS/grapesjs/assets/7035925/258ca917-f1bb-4a9c-9a51-9a408bad352b)

**Question:** When `+` button is clicked, the click event for `data-add-layer` is not getting triggered and it is not adding any layer to the background property.  I put a breakpoint in grapesJS library for `addLayer` function but the click is not getting there

Am i missing something ?

Thank you for your time,

**Comment by artf** [comment, marked as answer]
> the click event for data-add-layer is not getting triggered 

Not sure to what event are you referring. If you're creating your own style manager UI it's up to you to trigger the add of a new layer.
If you follow the example [here](https://grapesjs.com/docs/modules/Style-manager.html#custom-style-manager) you can see how it's done for the `stack` type.

<img width="729" alt="Screenshot 2023-10-08 at 13 26 03" src="https://github.com/GrapesJS/grapesjs/assets/11614725/b2bbb8a7-ac16-472a-83d0-beffe6f924b3">

**Comment by rajevar** [comment]
hi @artf ,
thank you for your reply. I thought declaring a button with `data-add-layer` would trigger click event. I came across this example from you on Jsfiddle and it helped me figure it out. https://jsfiddle.net/artur_arseniev/46kf7brn/
Thanks again for your time.
